language_specification/ldap.md

```diff
@@ -0,0 +1,219 @@
+
+This article explains how to configure the LDAP plugin and provides basic usage examples.
+
+## Configure
+
+### datasources.json
+
+In order to connect Lightblue to LDAP, you must first add a datasource to _datasources.json_.
+
+Multiple LDAP datasources can be used by adding additional blocks, each with a unique datasource and database name. 
+
+**Attributes:**
+ * **type** - Class path to the _DataSourceConfiguration_ implementation for LDAP. Should always be _com.redhat.lightblue.config.ldap.LdapDataSourceConfiguration_.
+ * **database** - Arbitrary name for the database, used by metadata to reference the target database.
+ * **bindableDn** - Similar to a username, this is the _dn_ used to authenticate into LDAP.
+ * **password** - Password associated with the _bindableDn_.
+ * **numberOfInitialConnections** - _(optional)_ The number of connections to initially establish when the pool is created. Defaults to 5.
+ * **maxNumberOfConnections** - _(optional)_ The maximum number of connections that should be maintained in the pool. Defaults to 10.
+ * **servers** - A list of _host_ and _port_ numbers to connect to. At least one must be provided. Multiples indicate mirrors that can be equally read from and written to.
+
+Example:
+```json
+{
+    "MyLdapDatasourceName": {
+        "type" : "com.redhat.lightblue.config.ldap.LdapDataSourceConfiguration",
+        "database" : "LdapDatabaseName",
+
+        "bindableDn" : "uid=admin,dc=example,dc=com",
+        "password" : "secret",
+        "numberOfInitialConnections" : 5,
+        "maxNumberOfConnections" : 10,
+        "servers" : [
+            {
+                "host" : "localhost",
+                "port" : "389"
+            }
+        ]
+    }
+}
+```
+
+### lightblue-metadata.json
+At this time LDAP is not a suitable backend metadata store and cannot be used in this capacity.
+
+### lightblue-crud.json
+Add the following block to the _controllers_ section in _lightblue-crud.json_. There really isn't anything to configure here, so the equivalent of a copy/paste should work.
+
+```json
+{
+   "controllers" : [
+      {
+        "backend" : "ldap",
+        "controllerFactory" : "com.redhat.lightblue.config.ldap.LdapControllerFactory"
+      }
+   ]
+}
+```
+
+## Metadata
+
+As with other plugins, the metadata is made up of two parts, the _entityInfo_ and _schema_. See the documentation on Metadata for more high level details on how these are used. Continue reading for LDAP specifics.
+
+More examples can be found in the **lightblue-ldap-integration-test** module.
+
+### entityInfo
+
+**Attributes:**
+* **backend** - Name of the datasource provider to use. Should always be "ldap".
+* **database** - Name of database to associate this entity too. Should match the _database_ attribute in _datasources.json_.
+* **basedn** - The base _dn_ for where the entities should live within the LDAP directory structure.
```

Comment:
`basedn` has the "d" lower case.
`bindableDn` has the "d" upper case.
Let's be consistent.